### DISCUSSION TITLE
location of docker-compose.yml to configure auto restart

### DISCUSSION THREAD

**Discussion by vincom** [mrlt8/docker-wyze-bridge#1111] [Q&A] [answered]
im a noob to docker. currently running docker-desktop on win11, got it running and working for wyze bridge and even configured to get the streams into blue iris as my nvr.
i just want the wyze container to auto start/restart on windows boot, but i can not locate the docker-compose.yml file to edit it and can not figure out how to create a new yml file to put into the container

**Comment by vincom** [comment, marked as answer]
figured it out. 
noob to github and docker but pc literate.
running docker desktop on win 11, got everything working as per instructions, but just wanted ability to auto restart wyze-bridge container if the win11 pc did a reboot, docker desktop was configured to start on windows boot. but could not find docker-compose.yml to edit the file for the container to  auto restart.

solution: i downloaded the wyze-bridge files from mrlt8 git library, unzipped it to its own directory, there is a docker-compose.yml file to edit, its located in the root of the wyze-bridge folder. ran "docker-compose up" in command prompt from within that directory .
everything working and wyze bridge can now restart on its own if computer reboots

**Comment by SplitMilky** [comment]
I'm here as well, and very lost.  I'm not new to linux or shells.  I am however new to WSL and docker.   I have my wyzecam running fine but i want to add/change some settings to the docker-compose.yml,  I cant find it anywhere.  

I've read multiple discussion about people not knowing where it is, and they seem to be solved but no one says where the damn yml file is.

I see you ran docker-compose up from a directory that you had the new downloaded yml file in, but does that mean you have to keep that file there or does docker pull it into its own area and save it?

I don't think I have been this lost using any software ever.

C:\Users\user\AppData\Local\Docker
\\wsl.localhost\docker-desktop
\\wsl.localhost\docker-desktop-data

There are some docker directories i found,  no where is there any docker-compose.yml

**Comment by vincom** [comment]
from what ive learned.
im using docker desktop on win10
the yml file can be anywhere locally, eg the pc with docker installed.
open a shell or command prompt aka terminal window from within the folder containing the yml file. how?
in that folder , nothing highlighted, right click + ctrl and select command prompt or power shell.
run  docker-compose up
if everything looks good stop the container from within docker desktop and run  docker-compose up -d
the yml file is just a config file to start everything, not used while running, without the file i can stop and restart docker desktop w/out issues.
this is what ive learned, please feel free to correct me if im wrong

**Comment by SplitMilky** [comment]
Yes this helps,  thank you.  

What I didn't realize is docker-compose uses the yml file to make a container out of an image.   (I think)

You can also use this method to change the settings of the container by running it again with an updated yml.  (I think)

Then after satisfied like you said use the -d to run in detached mode.

**Comment by vincom** [comment]
q.  What I didn't realize is docker-compose uses the yml file to make a container out of an image. (I think)?
a. yes

You can also use this method to change the settings of the container by running it again with an updated yml. (I think)?
yes

Then after satisfied like you said use the -d to run in detached mode.
bingo!
